### PR TITLE
Add distributed unit tests about text_classification/simnet-bow/ctr

### PR DESCRIPTION
**Pull request by seiriosPlus**
1.  fix test classification about #12248

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_dist_base.py [hunk 1]
```diff
@@ -155,28 +160,36 @@ def setUp(self):
...
         ps1_cmd = "%s %s pserver %s 0 %s %d TRUE %s" % \
-            (self._python_interp, model_file, self._ps_endpoints, ps1_ep,
-             self._trainers, sync_mode_str)
+                  (self._python_interp, model_file, self._ps_endpoints, ps1_ep,
+                   self._trainers, sync_mode_str)
 
         ps0_pipe = subprocess.PIPE
         ps1_pipe = subprocess.PIPE
         if check_error_log:
+            required_envs["GLOG_v"] = "7"
```
Comment (typhoonzero) [inline python/paddle/fluid/tests/unittests/test_dist_base.py]: remove debug log levels

File: python/paddle/fluid/tests/unittests/test_dist_inference_save_load.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        required_envs = {
+            "PATH": os.getenv("PATH"),
+            "PYTHONPATH": os.getenv("PYTHONPATH", ""),
+            "LD_LIBRARY_PATH": os.getenv("LD_LIBRARY_PATH", ""),
+            "FLAGS_fraction_of_gpu_memory_to_use": "0.15",
+            "FLAGS_cudnn_deterministic": "1"
+        }
+
+        if check_error_log:
+            required_envs["GLOG_v"] = "7"
```
Comment (typhoonzero) [inline python/paddle/fluid/tests/unittests/test_dist_inference_save_load.py]: No debug log here.

File: python/paddle/fluid/tests/unittests/test_dist_inference_save_load.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+        dirname = "/tmp/simnet.infer.model"
+        if trainer_id == 0:
+            self._save_model(dirname, [], [predict])
+
+    def check_with_save_inference(self,
+                                  model_file,
+                                  delta=1e-3,
+                                  check_error_log=False):
+        # *ATTENTION* THIS TEST NEEDS AT LEAST 2GPUS TO RUN
```
Comment (typhoonzero) [inline python/paddle/fluid/tests/unittests/test_dist_inference_save_load.py]: Try do not copy and paste code.

File: python/paddle/fluid/tests/unittests/test_dist_inference_save_load.py [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+                    (self._python_interp, model_file,
+                     "127.0.0.1:1234", "127.0.0.1:1234", 1, sync_mode_str)
+        if not check_error_log:
+            local_proc = subprocess.Popen(
+                local_cmd.split(" "),
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                env=env_local)
+        else:
+            print("trainer cmd:", local_cmd)
```
Comment (typhoonzero) [inline python/paddle/fluid/tests/unittests/test_dist_inference_save_load.py]: delete this line

File: python/paddle/fluid/tests/unittests/test_dist_base.py (hunk 1 above)
Comment (seiriosPlus) [reply python/paddle/fluid/tests/unittests/test_dist_base.py]: done

File: python/paddle/fluid/tests/unittests/test_dist_inference_save_load.py (hunk 2 above)
Comment (seiriosPlus) [reply python/paddle/fluid/tests/unittests/test_dist_inference_save_load.py]: I have deleted ```test_dist_inference_save_load.py```, will rewrite it in another PR.

File: python/paddle/fluid/tests/unittests/test_dist_text_classification.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+# limitations under the License.
+
+from __future__ import print_function
+import unittest
+from test_dist_base import TestDistBase
+
+
+class TestDistTextClassification2x2(TestDistBase):
+    def _setup_config(self):
+        self._sync_mode = True
```
Comment (gongweibao) [inline python/paddle/fluid/tests/unittests/test_dist_text_classification.py:23]: Do we need to add aync test?

File: python/paddle/fluid/tests/unittests/test_dist_simnet_bow.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+# limitations under the License.
+
+from __future__ import print_function
+import unittest
+from test_dist_base import TestDistBase
+
+
+class TestDistSimnetBow2x2(TestDistBase):
+    def _setup_config(self):
+        self._sync_mode = True
```
Comment (gongweibao) [inline python/paddle/fluid/tests/unittests/test_dist_simnet_bow.py:52]: Do we need to add aync test?

File: python/paddle/fluid/tests/unittests/dist_simnet_bow.py [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+
+    import os
+    os.environ['CPU_NUM'] = '1'
+    os.environ['USE_CUDA'] = 'FALSE'
+
+    os.environ["IS_DISTRIBUTED"] = '0'
+    os.environ["IS_SPARSE"] = '0'
+    runtime_main(TestDistSimnetBow2x2)
+
+#    os.environ["IS_DISTRIBUTED"] = '0'
```
Comment (gongweibao) [inline python/paddle/fluid/tests/unittests/dist_simnet_bow.py]: Is it useful?

Comment (seiriosPlus) [reply python/paddle/fluid/tests/unittests/dist_simnet_bow.py]: done

File: python/paddle/fluid/tests/unittests/test_dist_simnet_bow.py (hunk 6 above)
Comment (seiriosPlus) [reply python/paddle/fluid/tests/unittests/test_dist_simnet_bow.py:52]: done

File: python/paddle/fluid/tests/unittests/test_dist_text_classification.py (hunk 5 above)
Comment (seiriosPlus) [reply python/paddle/fluid/tests/unittests/test_dist_text_classification.py:23]: done

**Review by jacquesqiao — APPROVED** [review]
LGTM!

**Pull request merged**